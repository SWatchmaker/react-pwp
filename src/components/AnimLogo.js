import React from "react";

class AnimLogo extends React.Component {
  constructor() {
    super();
    this.palabra1 = "Seba";
    this.palabra2 = "Weidmann";
    this.state = {
      endLeave: true,
    };
  }

  componentDidMount() {
    this.logo1 = document.getElementsByClassName("logo-1");
    this.logo2 = document.getElementsByClassName("logo-2");
    setTimeout(() => {
      this.expand();
      setTimeout(() => {
        this.contract();
      }, 1000);
    }, 500);
  }

  expand = () => {
    this.inFlag = true;
    this.setState(() => {
      return {
        endLeave: false,
      };
    });
    this.logo1[0].textContent += "_";

    const timer = setInterval(() => {
      try {
        this.animflag = false;
        if (this.inFlag) {
          const largo1 = this.logo1[0].textContent.replace(/_/g, "").length;
          let paso2 = false;

          if (largo1 < this.palabra1.length) {
            this.logo1[0].textContent = this.logo1[0].textContent.replace(
              /_/g,
              ""
            );
            this.logo1[0].textContent += this.palabra1[largo1] + "_";
          } else {
            this.logo1[0].textContent = this.logo1[0].textContent.replace(
              /_/g,
              ""
            );
            paso2 = true;
            this.logo2[0].textContent += "_";
          }

          const largo2 = this.logo2[0].textContent.replace(/_/g, "").length;
          if (largo2 < this.palabra2.length && paso2) {
            this.logo2[0].textContent = this.logo2[0].textContent.replace(
              /_/g,
              ""
            );
            this.logo2[0].textContent += this.palabra2[largo2] + "_";
          } else if (paso2) {
            clearInterval(timer);
            this.logo2[0].textContent = this.logo2[0].textContent.replace(
              /_/g,
              ""
            );
            const timer2 = setInterval(() => {
              if (this.logo2[0].textContent.indexOf("_") > 0 && this.inFlag) {
                this.logo2[0].textContent = this.logo2[0].textContent.replace(
                  /_/g,
                  ""
                );
              } else if (
                this.logo2[0].textContent.indexOf(" ") &&
                this.inFlag
              ) {
                this.logo2[0].textContent = this.logo2[0].textContent + "_";
              } else {
                clearInterval(timer2);
              }
            }, 500);
          }
        } else {
          clearInterval(timer);
        }
      } catch (e) {
        console.log(e);
        clearInterval(timer);
      }
    }, 60);
  };

  contract = () => {
    this.inFlag = false;
    const timer = setInterval(() => {
      try {
        if (!this.inFlag) {
          let paso2 = false;

          if (this.logo2[0].textContent.indexOf("_") === -1) {
            this.logo2[0].textContent = this.logo2[0].textContent + "_";
          }
          const largo2 = this.logo2[0].textContent.replace(/_/g, "").length;
          if (largo2 > 1) {
            this.logo2[0].textContent =
              this.logo2[0].textContent.slice(0, largo2 - 1) + "_";
          } else {
            this.logo2[0].textContent = this.logo2[0].textContent.replace(
              /_/g,
              ""
            );
            this.logo1[0].textContent = this.logo1[0].textContent + "_";
            paso2 = true;
          }

          const largo = this.logo1[0].textContent.replace(/_/g, "").length;
          if (largo > 1 && paso2) {
            this.logo1[0].textContent =
              this.logo1[0].textContent.slice(0, largo - 1) + "_";
          } else if (paso2) {
            this.logo1[0].textContent = this.logo1[0].textContent.replace(
              /_/g,
              ""
            );
            this.animflag = true;
            this.setState(() => {
              return {
                endLeave: true,
              };
            });
            clearInterval(timer);
          }
        }
      } catch (e) {
        console.log(e);
        clearInterval(timer);
      }
    }, 60);
  };

  render() {
    return (
      <div
        className={`navbar-brand logo-anim border-anim animate__delay-1s animate__animated animate__fadeInDown ${
          !this.state.endLeave ? " in-anim" : ""
        }`}
        onMouseEnter={this.expand}
        onMouseLeave={this.contract}
      >
        <span className="logo-1">S</span>
        <span className="logo-2">W</span>
      </div>
    );
  }
}

export default AnimLogo;
