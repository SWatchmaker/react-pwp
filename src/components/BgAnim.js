import React, { useContext, useEffect, useState } from "react";
import ScrollContext from "../contexts/scrollContext";

let BgAnim = () => {
  const scrollY = useContext(ScrollContext);
  const [bg1Top, setBg1Top] = useState(0);
  const [bg1Height, setBg1Height] = useState(0);
  const [bgElement, setBgElement] = useState("motivationBox");
  useEffect(() => {
    if (scrollY <= 900 && bgElement != "motivationBox") {
      bg1Change("motivationBox");
      setBgElement("motivationBox");
    } else if (scrollY > 900 && bgElement != "findBox") {
      bg1Change("findBox");
      setBgElement("findBox");
    }
  });

  const bg1Change = (elementId) => {
    let top = document.getElementById(elementId).offsetTop - 150;
    if (Math.abs(top - bg1Top) >= 20) {
      let bg1Height = document.getElementById(elementId).offsetHeight + 300;
      setBg1Top(top);
      setBg1Height(bg1Height);
    }
  };

  return (
    <div className="bgContainer">
      <div
        className={`bgAnim1 ${scrollY > 180 ? "bgAnimIn" : "bgAnimOut"} ${
          scrollY > 900 && "bgAnim1-1"
        }`}
        style={{ top: bg1Top, height: bg1Height }}
      ></div>
      <div
        className={`bgAnim2 ${scrollY > 180 ? "bgAnimIn" : "bgAnimOut"} ${
          scrollY > 900 && "bgAnim2-1"
        }`}
        style={{ top: bg1Top, height: bg1Height }}
      ></div>
    </div>
  );
};

export default BgAnim;
