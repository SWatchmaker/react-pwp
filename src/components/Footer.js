import React, { useState, useEffect } from "react";
import Loader from "./Loader";

const Footer = props => {
  const [loadInit, setLoadInit] = useState(false);
  const [show, setShow] = useState(false);

  useEffect(() => {
    if (
      window.innerHeight + window.scrollY >= document.body.offsetHeight - 100 &&
      props.scrollY > 500
    ) {
      setShow(true);
      if (!loadInit) {
        setLoadInit(true);
      }
    } else {
      setShow(false);
    }
  });

  return (
    <footer id="footer" className={`footer ${show && "footerBg"}`}>
      {loadInit && (
        <div className="footer__content content-container">
          <div
            className={`footer__atomContainer animate__animated animate__fast ${
              show ? "animate__fadeInUp" : "animate__fadeOutDown"
            }`}
          >
            <Loader />
          </div>
          <div className="footer__info">
            <div className="footer__contact">
              <span
                className={`footer__contactTitle animate__animated ${
                  show ? "animate__fadeInLeft" : "animate__fadeOutRight"
                }`}
              >
                MIS REDES
              </span>
              <div className="footer__data">
                <a
                  href="https://www.linkedin.com/in/sebaweidmann/"
                  target="_blank"
                  className={`animate__animated animate__delay-1s ${
                    show ? "animate__fadeInLeft" : "animate__fadeOutRight"
                  }`}
                  style={{
                    "--animate-delay": "0.4s"
                  }}
                >
                  <i
                    className="fab fa-linkedin-in"
                    style={{
                      color: "#0077b5"
                    }}
                  ></i>
                  <span>Conoce mi CV</span>
                </a>
              </div>
              <div className="footer__data">
                <a
                  href="https://github.com/SWatchmaker"
                  target="_blank"
                  className={`animate__animated animate__delay-1s ${
                    show ? "animate__fadeInLeft" : "animate__fadeOutRight"
                  }`}
                  style={{
                    "--animate-delay": "0.8s"
                  }}
                >
                  <i
                    className="fab fa-github"
                    style={{
                      color: "#6e5494"
                    }}
                  ></i>
                  <span>Mis experimentos</span>
                </a>
              </div>
              <div className="footer__data">
                <a
                  href="mailto:sebastian.weidmann.l@gmail.com"
                  target="_blank"
                  className={`animate__animated animate__delay-1s ${
                    show ? "animate__fadeInLeft" : "animate__fadeOutRight"
                  }`}
                  style={{
                    "--animate-delay": "1.2s"
                  }}
                >
                  <i
                    className="far fa-envelope"
                    style={{
                      color: "#ea4335"
                    }}
                  ></i>
                  <span>Escríbeme!</span>
                </a>
              </div>
              <div className="footer__data">
                <a
                  href="https://wa.link/io5hce"
                  target="_blank"
                  className={`animate__animated animate__delay-1s ${
                    show ? "animate__fadeInLeft" : "animate__fadeOutRight"
                  }`}
                  style={{
                    "--animate-delay": "1.4s"
                  }}
                >
                  <i
                    className="fab fa-whatsapp"
                    style={{
                      color: "#25d366"
                    }}
                  ></i>
                  <span>Whatsapéame!</span>
                </a>
              </div>
            </div>
            <div className="footer__contact--right">
              <span
                className={`footer__contactTitle animate__animated ${
                  show ? "animate__fadeInRight" : "animate__fadeOutLeft"
                }`}
              >
                REDES INFORMALES
              </span>
              <div className="footer__data">
                <a
                  href="https://www.instagram.com/xoclomaster/"
                  target="_blank"
                  className={`animate__animated animate__delay-1s ${
                    show ? "animate__fadeInRight" : "animate__fadeOutLeft"
                  }`}
                  style={{
                    "--animate-delay": "0.4s"
                  }}
                >
                  <i
                    className="fab fa-instagram-square"
                    style={{
                      color: "#c13584"
                    }}
                  ></i>
                  <span>Instagram</span>
                </a>
              </div>
              <div className="footer__data">
                <a
                  href="https://www.facebook.com/SW.Watchmaker/"
                  target="_blank"
                  className={`animate__animated animate__delay-1s ${
                    show ? "animate__fadeInRight" : "animate__fadeOutLeft"
                  }`}
                  style={{
                    "--animate-delay": "0.8s"
                  }}
                >
                  <i
                    className="fab fa-facebook"
                    style={{
                      color: "#3b5998"
                    }}
                  ></i>
                  <span>Facebook</span>
                </a>
              </div>
              <div className="footer__data">
                <a
                  href="https://twitter.com/sawlrock"
                  target="_blank"
                  className={`animate__animated animate__delay-1s ${
                    show ? "animate__fadeInRight" : "animate__fadeOutLeft"
                  }`}
                  style={{
                    "--animate-delay": "1.2s"
                  }}
                >
                  <i
                    className="fab fa-twitter"
                    style={{
                      color: "#1da1f2"
                    }}
                  ></i>
                  <span>Twitter</span>
                </a>
              </div>

              <div className="footer__data">
                <a
                  href="https://steamcommunity.com/id/SWatchmaker/"
                  target="_blank"
                  className={`animate__animated animate__delay-1s ${
                    show ? "animate__fadeInRight" : "animate__fadeOutLeft"
                  }`}
                  style={{
                    "--animate-delay": "1.6s"
                  }}
                >
                  <i
                    className="fab fa-steam"
                    style={{
                      color: "#00adee"
                      //#C28F2C
                    }}
                  ></i>
                  <span>Steam</span>
                </a>
              </div>
            </div>
          </div>
        </div>
      )}
    </footer>
  );
};

export default Footer;
