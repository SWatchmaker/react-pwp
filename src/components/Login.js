import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

export const Login = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Boilerplate</h1>
      <p>Subt√≠tulo de la App</p>
      <button onClick={startLogin} className="button">
        Iniciar con Google
      </button>
    </div>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(Login);
