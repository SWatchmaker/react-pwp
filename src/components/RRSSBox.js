import React from "react";

class RRSSBox extends React.Component {
  constructor() {
    super();
    this.state = {
      anim: null,
    };
  }

  componentDidMount() {
    this.icons = document.getElementsByClassName("rsIcon");
    this.text = document.getElementById("contactText");
    for (var i = 0; i < this.icons.length; i++) {
      this.icons[i].addEventListener("animationend", (e) => {
        if (!this.state.anim && e.target.id === "gitIcon") {
          this.text.style.display = "inline-block";
          for (var i = 0; i < this.icons.length; i++) {
            this.icons[i].style.display = "none";
          }
        }
      });
    }
    this.text.addEventListener("animationend", (e) => {
      if (this.state.anim) {
        e.target.style.display = "none";
        for (var i = 0; i < this.icons.length; i++) {
          this.icons[i].style.display = "inline-block";
        }
      }
    });
  }

  inAnim = () => {
    this.setState(() => {
      return { anim: true };
    });
  };

  outAnim = () => {
    this.setState(() => {
      return { anim: false };
    });
  };

  render() {
    return (
      <div
        className="nav-item contact"
        onMouseEnter={this.inAnim}
        onMouseLeave={this.outAnim}
      >
        <div
          style={{ display: "none" }}
          className={`rsIcon animate__animated ${
            this.state.anim ? "animate__fadeInDown" : "animate__fadeOutRight"
          }`}
        >
          <i
            className="fab fa-whatsapp-square fa-2x"
            style={{
              color: "#25d366",
            }}
          ></i>
        </div>
        <div
          id="mailIcon"
          style={{ display: "none" }}
          className={`rsIcon animate__delay-1s animate__animated ${
            this.state.anim ? "animate__fadeInDown" : "animate__fadeOutRight"
          }`}
        >
          <i
            className="fas fa-envelope-square fa-2x"
            style={{
              color: "#ea4335",
            }}
          ></i>
        </div>
        <div
          id="ldIcon"
          style={{ display: "none" }}
          className={`rsIcon animate__delay-1s animate__animated ${
            this.state.anim ? "animate__fadeInDown" : "animate__fadeOutRight"
          }`}
        >
          <i
            className="fab fa-linkedin fa-2x"
            style={{
              color: "#0077b5",
            }}
          ></i>
        </div>
        <div
          id="gitIcon"
          style={{ display: "none" }}
          className={`rsIcon animate__delay-1s animate__animated ${
            this.state.anim ? "animate__fadeInDown" : "animate__fadeOutRight"
          }`}
        >
          <i
            className="rsIcon fab fa-github-square fa-2x"
            style={{
              color: "#6e5494",
            }}
          ></i>
        </div>
        <div
          id="contactText"
          style={{ display: "inline-block" }}
          className={`animate__animated ${
            this.state.anim ? "animate__fadeOutDown" : "animate__fadeInDown"
          }`}
        >
          <span>CONTACTO</span>
        </div>
      </div>
    );
  }
}

export default RRSSBox;
