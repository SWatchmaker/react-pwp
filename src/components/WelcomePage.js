import React, { useEffect, useState } from "react";
import Loader from "../components/Loader";
import { useLayoutEffect } from "react";

const WelcomPage = () => {
  const [nodePosition, setNodePos] = useState({
    top: Math.random() * 100,
    left: Math.random() * 100,
  });

  useEffect(() => {}, []);

  function getViewport() {
    var viewPortWidth;
    var viewPortHeight;

    // the more standards compliant browsers (mozilla/netscape/opera/IE7) use window.innerWidth and window.innerHeight
    if (typeof window.innerWidth != "undefined") {
      (viewPortWidth = window.innerWidth),
        (viewPortHeight = window.innerHeight);
    }

    // IE6 in standards compliant mode (i.e. with a valid doctype as the first line in the document)
    else if (
      typeof document.documentElement != "undefined" &&
      typeof document.documentElement.clientWidth != "undefined" &&
      document.documentElement.clientWidth != 0
    ) {
      (viewPortWidth = document.documentElement.clientWidth),
        (viewPortHeight = document.documentElement.clientHeight);
    }

    // older versions of IE
    else {
      (viewPortWidth = document.getElementsByTagName("body")[0].clientWidth),
        (viewPortHeight = document.getElementsByTagName("body")[0]
          .clientHeight);
    }
    return [viewPortWidth, viewPortHeight];
  }
  return (
    <div className="welcomePage">
      <Loader />
      <button className="welcomePage__inButton">BIENVENIDO</button>
      <div
        className="welcomePage__rollIcon"
        style={{ top: `${nodePosition.top}%`, left: `${nodePosition.left}%` }}
      >
        <i class="fab fa-html5"></i>
      </div>
      <div className="welcomePage__rollIcon">
        <i class="fab fa-node-js"></i>
      </div>
    </div>
  );
};

export default WelcomPage;
