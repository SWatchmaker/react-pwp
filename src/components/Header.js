import React from "react";
import { startLogout } from "../actions/auth";
import { connect } from "react-redux";
import AnimLogo from "./AnimLogo";
import RRSSBox from "./RRSSBox";

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <AnimLogo />
        <RRSSBox />
      </div>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout()),
});

export default connect(undefined, mapDispatchToProps)(Header);
