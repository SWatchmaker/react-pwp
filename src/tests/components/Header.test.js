import React from "react";
import { shallow } from "enzyme";
import { Header } from "../../components/Header";

test("Renderiza Header", () => {
  const wrapper = shallow(<Header startLogout={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test("Inicia Logout al clickera botÃ³n", () => {
  const startLogout = jest.fn();
  const wrapper = shallow(<Header startLogout={startLogout} />);
  wrapper.find("button").simulate("click");

  expect(startLogout).toHaveBeenCalled();
});
