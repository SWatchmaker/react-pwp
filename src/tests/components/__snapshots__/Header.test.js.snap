// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderiza Header 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Gastadero
        </h1>
      </Link>
      <button
        className="button button--link"
        onClick={[Function]}
      >
        Cerrar Sesi√≥n
      </button>
    </div>
  </div>
</header>
`;
